### PR TITLE
Update unit testing

### PR DESCRIPTION
**Pull request by shadowhand**
Upgrades PHPUnit and adds a `phpunit.xml.dist` file so that code coverage can be generated.

Also removes `composer.lock` from git control.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -21,13 +21,13 @@
     "license": "MIT",
     "name": "php-fp/php-fp-either",
     "scripts": {
-        "test": "vendor/bin/phpunit test"
+        "test": "phpunit test"
```
Comment (shadowhand) [inline composer.json:24]: Composer bin directory is [automagically resolved for scripts](https://getcomposer.org/doc/articles/scripts.md#writing-custom-commands).

**Comment by i-am-tom** [comment]
What's the reasoning behind removing `composer.lock`? I was under the impression that it was very much there to be committed!

**Comment by shadowhand** [comment]
https://getcomposer.org/doc/02-libraries.md#lock-file

The only reason to commit the lock file would be to enforce strict testing against dependency versions. This is not necessary for Either or (most) libraries, which want to be as compatible as possible. For projects, it is definitely recommended to commit `composer.lock`.

**Comment by i-am-tom** [comment]
Cool! Thanks for this, @shadowhand :)

**Pull request merged**